cmd/influxd/run.go

```diff
@@ -160,10 +159,34 @@ func (s *Node) closeSnapshotListener() error {
 	return err
 }
 
-func (cmd *RunCommand) Run(args ...string) error {
-	// Set up logger.
-	cmd.Logger = log.New(os.Stderr, "", log.LstdFlags)
+func (cmd *RunCommand) ParseConfig(path, hostname string) error {
+	var err error
+
+	// Parse configuration file from disk.
+	if path != "" {
+		cmd.config, err = ParseConfigFile(path)
+
+		if err != nil {
+			return fmt.Errorf("error parsing config %s - %s\n", path, err)
+		}
+		// Override config properties.
+		if hostname != "" {
+			cmd.config.Hostname = hostname
+		}
+
+		log.Printf("using config: %s\n", path)
+		return nil
+	}
 
+	cmd.config, err = NewTestConfig()
+	if err != nil {
+		return fmt.Errorf("error parsing default config: %s\n", err)
+	}
+	log.Println("no config provided, using default settings")
```

Comment:
Pure style: I think we should spell out `config` when messages are for users too.